bec_widgets/widgets/utility/logpanel/logpanel.py

```diff
@@ -93,19 +93,24 @@ def __init__(
         self._selected_services: set[str] | None = None
         self._set_formatter_and_update_filter(line_formatter)
         self._conn.register([MessageEndpoints.log()], None, self._process_incoming_log_msg)
+        self.destroyed.connect(self.unsub_from_redis)
```

Comment:
This is likely never triggered as the signal will be emitted right before the c++ object gets destroyed, so your class is likely already gone. In general, it is better to use the destroyed signal of other objects, e.g the parent object. 
If you inherit from BECConnector, a destroyed parent will automatically call the cleanup method.

Reply:
The issue is that occasionally the c++ object for `BecLogsQueue` was getting destroyed independently without its parent - I could replicate it with the designer, although it doesn't happen for me locally while using the widget in `bec` normally. This doesn't seem to be resolved by inheriting from `BECConnector` and moving the unsub into `cleanup()`.

Reply:
My guess would be that your change of line 419, i.e. setting the parent of BecLogsQueue not to the parent of the main widget but rather to the main widget itself should solve the problem of losing the underlying c++ object. Still, subscribing to the destroyed signal from within the qobject is likely not going to work. But maybe just test it with a debugger and see if you are actually triggering and receiving the signal.